### ISSUE TITLE
Variable values from previous kernel session are redisplayed in variable explorer after a restart

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode-jupyter#7126]
1. Open a notebook
2. Run `a = 1` in a cell
3. `a` appears in the variable explorer
4. Restart the kernel
5. Variable explorer is cleared
6. Run `a` in a new cell
7. 🐛 `a` will fail because the kernel was restarted, however the variable explorer updates to show the old, cached value of `a`

**Comment by joyceerhl** [comment]
It looks like this is because the KernelVariablesRequester only listens to INotebook.onKernelRestarted to clear its cached scripts import  https://github.com/microsoft/vscode-jupyter/blob/9719f6f65a70a00a89ebacc425fe0b844a1281e5/src/client/datascience/jupyter/pythonVariableRequester.ts#L196

And since we are no longer using INotebook.restartKernel (or at least trying to move away from it), that event doesn't get fired when a restart occurs.

**Comment by joyceerhl** [comment]
This is what you'll get in the error log 

```
Error 2021-08-13 14:34:17: Failure during variable extraction:
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-2-dcb72fef6c4b> in <module>
      1 _rwho_ls = get_ipython().run_line_magic('who_ls', '')
----> 2 print(_VSCODE_getVariableTypes(_rwho_ls))

NameError: name '_VSCODE_getVariableTypes' is not defined
```

Basically the query to pick up the new variables fails because the KernelVariablesRequester wasn't told about the restart

**Comment by joyceerhl** [comment]
Should be fixed with #7129
